### ISSUE TITLE
TrackballControls pinch to zoom behaves inconsistently with recent browser updates

### ISSUE THREAD

**Issue by mholladay** [mrdoob/three.js#14952]
##### Description of the problem

When using pinch to zoom with the TrackballControls (for example, at https://threejs.org/examples/misc_controls_trackball.html) as you zoom in both three.js and the browser are processing the touch controls. So the stats in the upper left and the title across the top disappear as they zoom off the screen and the size of the objects jumps around as both the 3d scene a the div are zoomed. 

After some investigation, it appears that browsers are starting set the "passive" flag associated with the touch listeners to "true" by default (see https://developer.mozilla.org/en-US/docs/Web/Events/touchstart). That causes the "preventDefault" call to get ignored, resulting in this strangeness.

I can work around the issue by updating the code at the end of TrackballControls.js to look like this. Still looking to see if I can find a cleaner solution.
```
	var passiveOffIfSupported = false;
	try {

		window.addEventListener("test", null, Object.defineProperty( {}, "passive", { get: function () { passiveOffIfSupported = { capture: false, passive: false }; } } ) );

	} 
	catch (error) {
	}

	this.dispose = function () {

		this.domElement.removeEventListener( 'contextmenu', contextmenu, false );
		this.domElement.removeEventListener( 'mousedown', mousedown, false );
		this.domElement.removeEventListener( 'wheel', mousewheel, false );

		this.domElement.removeEventListener( 'touchstart', touchstart, passiveOffIfSupported );
		this.domElement.removeEventListener( 'touchend', touchend, passiveOffIfSupported );
		this.domElement.removeEventListener( 'touchmove', touchmove, passiveOffIfSupported );

		document.removeEventListener( 'mousemove', mousemove, false );
		document.removeEventListener( 'mouseup', mouseup, false );

		window.removeEventListener( 'keydown', keydown, false );
		window.removeEventListener( 'keyup', keyup, false );

	};

	this.domElement.addEventListener( 'contextmenu', contextmenu, false );
	this.domElement.addEventListener( 'mousedown', mousedown, false );
	this.domElement.addEventListener( 'wheel', mousewheel, false );

	this.domElement.addEventListener( 'touchstart', touchstart, passiveOffIfSupported );
	this.domElement.addEventListener( 'touchend', touchend, passiveOffIfSupported );
	this.domElement.addEventListener( 'touchmove', touchmove, passiveOffIfSupported );
```

##### Three.js version

- [x] Dev
- [x] r96
- [ ] ...

##### Browser

- [ ] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer
- [x] Safari

##### OS

- [ ] All of them
- [x] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [x] iOS

Note: Only tested on Chrome (69.0.3497.100) and Safari in Windows 10 64-bit and iOS 11.4.1. I expect that it occurs across all OS and browsers that are changing the default for the "passive" flag per  https://developer.mozilla.org/en-US/docs/Web/Events/touchstart.

##### Hardware Requirements (graphics card, VR Device, ...)
N/A

**Comment by mholladay** [comment]
The new passive default for touch listeners currently only applies to the "window" and "document" objects. So you can also work around this problem by specifically setting up the controls to use the renderer.domElement. For example:
```controls = new THREE.TrackballControls( camera, renderer.domElement );```

(If using the misc_controls_trackball.html example you also need to initialize the renderer before doing this.)

However, if you want to use touch controls that are attached to the window or document rather than the renderer.domElement, you do need to use an approach similar to the one in the original description above.

Note that you can also reproduce this problem with orbit controls if you allow them to be attached to the document rather than the renderer.domElement.

**Comment by WestLangley** [comment]
We have recommended the use of this pattern:
```javascript
controls = new THREE.OrbitControls( camera, renderer.domElement );
```
Is there a use case for which that pattern would not be appropriate?

@mholladay Are you recommending a change in the library?

**Comment by mholladay** [comment]
That pattern works, but in cases where a renderer.domElement is not passed in, OrbitControls.js and TrackballControls.js fall back to attaching the event listeners to the "document", which runs into this problem.

For example, in misc_controls_trackball.html it uses:
```
controls = new THREE.TrackballControls( camera );
```

If you work from that example or any others that do not pass renderer.domElement, you end up with odd touch behaviors. It would be nice if the controls forced "passive" to false to prevent the problem -- but updating the examples to use the recommended pattern would probably be sufficient.

**Comment by Mugen87** [comment]
Can you please check of the changes of the following PR solve the problem: #14182

Also see: https://bugs.chromium.org/p/chromium/issues/detail?id=841282

**Comment by mholladay** [comment]
@Mugen87  - #14182 addresses the issue for Chrome running on Windows. However, the problem persists in Safari (at least in iOS 11) since Safari does not support the 'none' touchAction. Looking at the current WebKit development it doesn't look like they plan on supporting it any time soon, either. Additionally, in Chrome 68 on iOS 11 it results in an odd flashing behavior during pinch zoom.

Cleanest solution across platforms for what I'm working on is to pass in the "renderer.domElement" rather than allowing the controls to attach directly to "document". Since WebKit does not support the 'none' touchAction I believe a cross-platform solution when attaching to "document" would require forcing the passive flag to "false" when attaching the touch event listeners.

**Comment by schlamar** [comment]
I'm still having this issue in latest release. Debug console shows:

    OrbitControls.js:832 [Intervention] Unable to preventDefault inside passive event listener due to target being treated as passive. See https://www.chromestatus.com/features/5093566007214080

Adding `user-scalable=no` to meta data would be another workaround:

    <meta name="viewport" content="width=device-width, user-scalable=no">

I guess updating the examples to use the recommended pattern and issue a warning to the console if no domElement is passed would be the best solution.

BTW you can test this behaviour on Chrome with the device emulation mode. Ctrl + Shift + Mouse Click + Mouse move triggers this issue.

**Comment by Mugen87** [comment]
> However, the problem persists in Safari (at least in iOS 11) since Safari does not support the 'none' touchAction. Looking at the current WebKit development it doesn't look like they plan on supporting it any time soon, either. 

It seems support for `touch-action` was lately added to Webkit 🎉 

https://bugs.webkit.org/show_bug.cgi?id=133112

**Comment by Mugen87** [comment]
> Cleanest solution across platforms for what I'm working on is to pass in the "renderer.domElement" rather than allowing the controls to attach directly to "document".

This should be solved now via #17612
